### PR TITLE
minor: fixed TeamCity build violation

### PR DESCRIPTION
**Pull request by shashwatj07**
Leftover from #10073
See https://github.com/checkstyle/checkstyle/pull/10073#issuecomment-858387865

### CODE REVIEW

**Comment by strkkk** [comment]
@shashwatj07 since TC is not working for PRs, can you please verify locally that it fixes issue? I mean something like screens before/after

**Comment by shashwatj07** [comment]
@strkkk 
Before

After

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/PkgImportControl.java [hunk 1]
```diff
@@ -158,6 +158,7 @@ private static String encloseInGroup(String expression) {
      * @return a regex string.
      */
     private static String toRegex(String input) {
+        // noinspection DynamicRegexReplaceableByCompiledPattern
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/PkgImportControl.java:161]: it looks like that more common way is to do like
`@noinspection UseOfPropertiesAsHashtable` in javadocs

Comment (shashwatj07) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/PkgImportControl.java:161]:
```
/**
     * Converts a normal package name into a regex pattern by escaping all
     * special characters that may occur in a java package name.
     *
     * @param input the input string.
     * @return a regex string.
     */
    @SuppressWarnings("DynamicRegexReplaceableByCompiledPattern")
    private static String toRegex(String input) {
        return input.replace(DOT, DOT_REGEX);
    }
```
If you want to suppress for entire method, IntelliJ is suggesting this.

Comment (shashwatj07) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/PkgImportControl.java:161]: ```// noinspection DynamicRegexReplaceableByCompiledPattern``` suppresses it for one statement only

Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/PkgImportControl.java:161]: ok, lets keep it small

**Review by strkkk — APPROVED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/PkgImportControl.java (hunk 1 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/PkgImportControl.java:161]: @shashwatj07
Please move it to javadoc, we do not need to polite code with  comments

**Review by romani — CHANGES_REQUESTED** [review]
Items:

**Comment by shashwatj07** [comment]
> Please move it to javadoc, we do not need to polite code with comments

@romani please review now

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/PkgImportControl.java (hunk 1 above)
Comment (shashwatj07) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/PkgImportControl.java]: done

Comment (shashwatj07) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/PkgImportControl.java]: ```@noinspection DynamicRegexReplaceableByCompiledPattern``` didn't seem to work in my IDE. Please see.

If it's not working we can do this https://github.com/checkstyle/checkstyle/pull/10103#discussion_r649042275

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/PkgImportControl.java [hunk 2]
```diff
@@ -156,6 +156,7 @@ private static String encloseInGroup(String expression) {
      *
      * @param input the input string.
      * @return a regex string.
+     * @noinspection DynamicRegexReplaceableByCompiledPattern
      */
     private static String toRegex(String input) {
         return input.replace(DOT, DOT_REGEX);
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/PkgImportControl.java:162]:
```
    private static final String DOT = ".";
    private static final String DOT_REGEX = "\\.";
    private static final String DOT_ESCAPED = "\\\\.";
    private static final Pattern DOT_PATTERN2 = Pattern.compile(DOT_REGEX);
    public static void main(String... args) {
        final String string = "com.puppycrawl.tools.checkstyle";
        System.out.println(string);
        System.out.println(DOT_PATTERN2.matcher(string).replaceAll(DOT_ESCAPED));
        System.out.println(string.replace(DOT, DOT_REGEX));
    }
```

```
...
com.puppycrawl.tools.checkstyle
com\.puppycrawl\.tools\.checkstyle
com\.puppycrawl\.tools\.checkstyle

Process finished with exit code 0
```

**Review by romani — CHANGES_REQUESTED** [review]
https://stackoverflow.com/a/19745865/1015848

code should be updated:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/PkgImportControl.java (hunk 2 above)
Comment (shashwatj07) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/PkgImportControl.java]: Done

**Review by romani — APPROVED** [review]

**Pull request merged**